validator/src/main/java/com/amazon/aoc/validators/AbstractStructuredLogValidator.java

```diff
@@ -90,56 +91,59 @@ public void validate() throws Exception {
       fetchAndValidateLogs(startTime);
       checkResult();
     });
-    log.info("[ContainerInsight] finish validation successfully");
-  }
-
-  protected Set<String> getValidatingLogStreamNames() {
-    return new HashSet<>();
+    log.info("[StructuredLogValidator] finish validation successfully");
   }
 
   protected void fetchAndValidateLogs(Instant startTime) throws Exception {
-    for (String logStreamName : getValidatingLogStreamNames()) {
+    for (String logStreamName : logStreamNames) {
       List<OutputLogEvent> logEvents = cloudWatchService.getLogs(logGroupName, logStreamName,
               startTime.toEpochMilli(), QUERY_LIMIT);
       if (logEvents.isEmpty()) {
         throw new BaseException(
                 ExceptionCode.LOG_FORMAT_NOT_MATCHED,
-                String.format("[ContainerInsight] no logs found under log stream %s",
+                String.format("[StructuredLogValidator] no logs found under log stream %s",
                         logStreamName));
       }
       for (OutputLogEvent logEvent : logEvents) {
         validateJsonSchema(logEvent.getMessage());
       }
-    } 
+    }
+  }
+
+  private void checkResult() throws BaseException {
+    if (validateJsonSchema.isEmpty()) {
+      return;
+    }
+    String[] failedTargets = new String[validateJsonSchema.size()];
+    int i = 0;
+    for (String key : validateJsonSchema.keySet()) {
+      failedTargets[i] = key;
+      i++;
+    }
+    throw new BaseException(
+            ExceptionCode.LOG_FORMAT_NOT_MATCHED,
+            String.format("[StructuredLogValidator] log structure validation failed for %s",
+                    StringUtils.join(",", failedTargets)));
   }
 
   protected void validateJsonSchema(String logEventMsg) throws Exception {
     JsonNode logEventNode = mapper.readTree(logEventMsg);
-    JsonSchema jsonSchema = findJsonSchemaForValidation(logEventNode);
+    String key = getJsonSchemaMappingKey(logEventNode);
```

Comment:
Can we provide a way to toggle the report level?  When missing field, the validator does not throw exception in default level and it's hard to debug. The draw back is once enabled it will print a lot of useless information as most time only a few log in the stream matches the full schema. It's mainly for debugging during development.

Reply:
Supported the exception level customization. I agree that we possibly can't benefit from it for CI job, but I feel it would be useful when unit testing Validators. Nice to have it.